### PR TITLE
RTL layout compatibility: bridges list on room settings page

### PR DESCRIPTION
**Pull request by luixxiul**
Addresses https://github.com/vector-im/element-web/issues/14520 (#8217)

#### Current implementation:

LTR layout
before

RTL layout
before2

The icon is moved to the left side because of the `float: right;`. This PR fixes the bug by replacing float with flexbox (note: you could otherwise set `float: left` for the RTL layout by either replacing the current padding setting with a logical one and setting a specific className for the layout to a higher element). 

after1
after2

The position of the avatar inside pill is out of scope of this PR. It should be taken care of by #8217.

Multiple bridges:

after3
after4

Signed-off-by: Suguru Hirahara <luixxiul@users.noreply.github.com>

type: task

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://pr8243--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: res/css/views/dialogs/_RoomSettingsDialogBridges.scss [hunk 1]
```diff
@@ -22,84 +22,95 @@ limitations under the License.
...
-            color: $muted-fg-color;
-            margin-bottom: 0;
-            overflow-y: visible;
-            text-overflow: ellipsis;
-            white-space: normal;
-            padding: 0;
-
-            > li {
-                padding: 0;
-                border: 0;
```
Comment (luixxiul) [inline res/css/views/dialogs/_RoomSettingsDialogBridges.scss:102]: These can be removed without causing a regression thanks to proper nesting

File: src/components/views/settings/BridgeTile.tsx [hunk 2]
```diff
@@ -115,7 +115,7 @@ export default class BridgeTile extends React.PureComponent<IProps> {
         if (protocol.avatar_url) {
             const avatarUrl = mediaFromMxc(protocol.avatar_url).getSquareThumbnailHttp(64);
 
-            networkIcon = <BaseAvatar className="protocol-icon"
+            networkIcon = <BaseAvatar className="protocolIcon"
```
Comment (t3chguy) [inline src/components/views/settings/BridgeTile.tsx:118]: given you're fixing this className anyway, can you scope it to the file

File: src/components/views/settings/BridgeTile.tsx [hunk 3]
```diff
@@ -146,19 +146,19 @@ export default class BridgeTile extends React.PureComponent<IProps> {
...
+            <div className="column_data">
+                <h3 className="column_data_protocolName">{ protocolName }</h3>
+                <p className="column_data_details workspace_channel_details">
                     { networkItem }
                     <span className="channel">{ _t("Channel: <channelLink/>", {}, {
                         channelLink: () => channelLink,
                     }) }</span>
                 </p>
-                <ul className="metadata">
+                <ul className="column_data_metadata metadata">
```
Comment (t3chguy) [inline src/components/views/settings/BridgeTile.tsx:161]: same with all of these classes please

File: res/css/views/dialogs/_RoomSettingsDialogBridges.scss [hunk 4]
```diff
@@ -22,84 +22,99 @@ limitations under the License.
...
+
+                .column_data_protocolName {
+                    margin-top: 0;
+                    font-size: 16pt;
+                    color: $primary-content;
+                }
+
+                .workspace_channel_details {
+                    color: $primary-content;
+                    font-weight: 600;
```
Comment (t3chguy) [inline res/css/views/dialogs/_RoomSettingsDialogBridges.scss:103]:
```suggestion
                    font-weight: $font-semi-bold;
```

**Review by t3chguy — COMMENTED** [review]
Looks sane otherwise

File: res/css/_common.scss [hunk 5]
```diff
@@ -227,7 +227,7 @@ legend {
 
 /*** panels ***/
 .dark-panel {
-    background-color: $dark-panel-bg-color;
+    background-color: #111;
```
Comment (t3chguy) [inline res/css/_common.scss:230]: This looks unintentional

Comment (luixxiul) [reply res/css/_common.scss:230]: yes, it was a mistake.

**Review by t3chguy — APPROVED** [review]
Looks sane to me, thanks!

**Comment by t3chguy** [comment]
Can you please pull latest develop

**Comment by luixxiul** [comment]
Done, but it's stuck like this:

Screenshot_2022-04-19_11-14-37

I'm not really sure what that means...

**Pull request merged**